homeassistant/components/androidtv/entity.py

```diff
@@ -73,7 +73,7 @@ async def _adb_exception_catcher(
                     func.__name__,
                 )
                 return None
-            except self.exceptions as err:
+            except Exception as err:  # pylint: disable=broad-except
```

Comment:
Do you really need the broad exception catch here? 

Couldn't you just list the ones that you are aware of (perhaps those defined in constants in `__init__.py`), so that python is still able to raise the exceptions that you really didn't expect?

Reply:
The reason for broad exception is that we need to set entity as disconnected in any case. The original implementation was to throw the exception in case of unknown exception, but in that case the entity class just catch the generic Exception and log a message, leaving entity status as is, that is not what we want.

Reply:
Ok, so the main thing you wanted was not to re-raise the exception at the end of the broad catch?

Could you add a log message there to state 'unexpected exception', then return?

I don't see a big issue here but for me, grouping foreseen behaviour with completely unforeseen behaviour can make fault finding harder in the future. 

It's also difficult for future maintainers to know what you thought the 'normal' exceptions would be if everything is in a catch all.

Reply:
Ok, so I restored the part related to dedicated exception and added a specific log for generic, unexpected exception